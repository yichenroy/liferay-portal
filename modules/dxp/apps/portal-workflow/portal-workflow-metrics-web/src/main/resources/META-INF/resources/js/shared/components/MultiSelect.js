import Icon from './Icon';
import React from 'react';

/**
 * @class
 * @memberof shared/components
 * */
export default class MultiSelect extends React.Component {
	constructor({data, selectedTags}) {
		super({data, selectedTags});

		this.state = {
			active: false,
			data,
			searchKey: '',
			selectedTags,
			verticalIndex: -1
		};

		this.addTag = this.addTag.bind(this);
		this.hideDropList = this.hideDropList.bind(this);
		this.onSearch = this.onSearch.bind(this);
		this.removeTag = this.removeTag.bind(this);
		this.showDropList = this.showDropList.bind(this);
	}

	get dataFiltred() {
		const {data, searchKey, selectedTags} = this.state;
		const term = searchKey.toLowerCase().trim();

		return data
			.filter(tag => selectedTags.indexOf(tag) === -1)
			.filter(tag => term === '' || tag.toLowerCase().indexOf(term) > -1);
	}

	addTag(event) {
		const tag = event.currentTarget.getAttribute('data-tag');
		const {selectedTags} = this.state;

		this.setState({
			selectedTags: [...selectedTags, tag]
		});
	}

	hideDropList() {
		this.setState({active: false});
	}

	onSearch(event) {
		const dataFiltred = this.dataFiltred;
		const {
			keyCode,
			target: {value: searchKey}
		} = event;
		let {selectedTags, verticalIndex} = this.state;

		if ([38, 40].indexOf(keyCode) > -1) {
			const dataCount = dataFiltred.length;

			verticalIndex = keyCode === 40 ? verticalIndex + 1 : verticalIndex - 1;
			verticalIndex =
				verticalIndex < 0
					? 0
					: verticalIndex + 1 > dataCount
						? dataCount - 1
						: verticalIndex;

			this.setState({verticalIndex});
		}
		else if (keyCode === 13 && verticalIndex > -1) {
			selectedTags = [...selectedTags, dataFiltred[verticalIndex]];
			this.setState({
				searchKey: '',
				selectedTags,
				verticalIndex: -1
			});
		}
		else {
			this.setState({
				searchKey,
				verticalIndex: -1
			});
		}
	}

	removeTag(event) {
		const tagIndex = Number(event.currentTarget.getAttribute('data-tag-index'));
		const {selectedTags} = this.state;

		selectedTags.splice(tagIndex, 1);
		this.setState({
			selectedTags
		});
	}

	showDropList() {
		this.setState({active: true});
	}

	render() {
		const {active, selectedTags, verticalIndex} = this.state;

		const tagRender = (text, tagIndex) => (
			<span
				className="label label-dismissible label-secondary"
				key={tagIndex}
				tabIndex="-1"
			>
				<span className="label-item label-item-expand">{text}</span>
				<span className="label-item label-item-after">
					<button
						aria-label="Close"
						className="close"
						data-tag-index={`${tagIndex}`}
						onClick={this.removeTag}
						type="button"
					>
						<Icon iconName="times" />
					</button>
				</span>
			</span>
		);

		const className = `${
			active ? 'is-active' : ''
		} align-items-start form-control form-control-tag-group multi-select-wrapper`;

		return (
			<div
				className={className}
				onDoubleClick={this.hideDropList}
				onFocus={this.showDropList}
			>
				<div className="col-11 p-0">
					{selectedTags.map((tag, index) => tagRender(tag, index))}
					<input
						className="form-control-inset"
						onKeyUp={this.onSearch}
						type="text"
					/>
				</div>
				<div className="col-1 mt-1">
					<Icon iconName="caret-double" />
				</div>
				{active && (
					<div className="drop-suggestion dropdown-menu" tabIndex="-1">
						<ul className="list-unstyled">
							{this.dataFiltred.map((tag, index) => (
								<li data-tag={`${tag}`} key={index} onClick={this.addTag}>
									<a
										className={`dropdown-item ${
											index === verticalIndex ? 'active' : ''
										}`}
										href={`#${index}`}
									>
										{tag}
									</a>
								</li>
							))}
						</ul>
					</div>
				)}
			</div>
		);
	}
}