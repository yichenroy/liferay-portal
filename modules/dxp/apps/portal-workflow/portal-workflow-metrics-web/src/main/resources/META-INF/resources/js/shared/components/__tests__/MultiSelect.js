import MultiSelect from '../MultiSelect';
import React from 'react';

const data = [
	'ops',
	'message',
	'meeting',
	'assist',
	'not real',
	'advocate',
	'Maybe',
	'much',
	'so much',
	'to do any',
	'a long component text example to test its overflow behavior.'
];
const selectedTags = ['ops', 'message', 'advocate', 'Maybe'];

test('Should test component render', () => {
	const component = shallow(
		<MultiSelect data={data} selectedTags={selectedTags} />
	);

	expect(component).toMatchSnapshot();
});

test('Should show the dropdown list', () => {
	const component = shallow(
		<MultiSelect data={data} selectedTags={selectedTags} />
	);

	const instance = component.instance();

	instance.showDropList();
	expect(component).toMatchSnapshot();
});

test('Should hide the dropdown list', () => {
	const component = shallow(
		<MultiSelect data={data} selectedTags={selectedTags} />
	);

	const instance = component.instance();

	instance.hideDropList();
	expect(component).toMatchSnapshot();
});

test('Should add a tag', () => {
	const component = shallow(
		<MultiSelect data={data} selectedTags={selectedTags} />
	);

	const instance = component.instance();

	instance.addTag({
		currentTarget: {
			getAttribute: () => 'test'
		}
	});
	expect(component).toMatchSnapshot();
});

test('Should remove a tag', () => {
	const component = shallow(
		<MultiSelect data={data} selectedTags={selectedTags} />
	);

	const instance = component.instance();

	instance.removeTag({
		currentTarget: {
			getAttribute: () => '0'
		}
	});
	expect(component).toMatchSnapshot();
});

test('Should test keydown press', () => {
	const component = shallow(
		<MultiSelect data={data} selectedTags={selectedTags} />
	);

	const instance = component.instance();

	instance.onSearch({
		keyCode: 38,
		target: { value: '' }
	});
	expect(component).toMatchSnapshot();
});

test('Should test keyup press', () => {
	const component = shallow(
		<MultiSelect data={data} selectedTags={selectedTags} />
	);

	const instance = component.instance();

	instance.onSearch({
		keyCode: 40,
		target: { value: '' }
	});
	expect(component).toMatchSnapshot();
});

test('Should test keyenter press', () => {
	const component = shallow(
		<MultiSelect data={data} selectedTags={selectedTags} />
	);

	const instance = component.instance();

	instance.onSearch({
		keyCode: 38,
		target: { value: '' }
	});
	instance.onSearch({
		keyCode: 13,
		target: { value: '' }
	});
	expect(component).toMatchSnapshot();
});

test('Should search', () => {
	const component = shallow(
		<MultiSelect data={data} selectedTags={selectedTags} />
	);

	const instance = component.instance();

	instance.onSearch({
		keyCode: 84,
		target: { value: 'test' }
	});
	expect(component).toMatchSnapshot();
});