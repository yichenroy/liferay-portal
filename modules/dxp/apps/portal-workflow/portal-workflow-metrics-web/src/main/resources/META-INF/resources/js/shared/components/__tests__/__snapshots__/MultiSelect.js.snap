// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should add a tag 1`] = `
<div
  className=" align-items-start form-control form-control-tag-group multi-select-wrapper"
  onDoubleClick={[Function]}
  onFocus={[Function]}
>
  <div
    className="col-11 p-0"
  >
    <span
      className="label label-dismissible label-secondary"
      key="0"
      tabIndex="-1"
    >
      <span
        className="label-item label-item-expand"
      >
        ops
      </span>
      <span
        className="label-item label-item-after"
      >
        <button
          aria-label="Close"
          className="close"
          data-tag-index="0"
          onClick={[Function]}
          type="button"
        >
          <Icon
            iconName="times"
          />
        </button>
      </span>
    </span>
    <span
      className="label label-dismissible label-secondary"
      key="1"
      tabIndex="-1"
    >
      <span
        className="label-item label-item-expand"
      >
        message
      </span>
      <span
        className="label-item label-item-after"
      >
        <button
          aria-label="Close"
          className="close"
          data-tag-index="1"
          onClick={[Function]}
          type="button"
        >
          <Icon
            iconName="times"
          />
        </button>
      </span>
    </span>
    <span
      className="label label-dismissible label-secondary"
      key="2"
      tabIndex="-1"
    >
      <span
        className="label-item label-item-expand"
      >
        advocate
      </span>
      <span
        className="label-item label-item-after"
      >
        <button
          aria-label="Close"
          className="close"
          data-tag-index="2"
          onClick={[Function]}
          type="button"
        >
          <Icon
            iconName="times"
          />
        </button>
      </span>
    </span>
    <span
      className="label label-dismissible label-secondary"
      key="3"
      tabIndex="-1"
    >
      <span
        className="label-item label-item-expand"
      >
        Maybe
      </span>
      <span
        className="label-item label-item-after"
      >
        <button
          aria-label="Close"
          className="close"
          data-tag-index="3"
          onClick={[Function]}
          type="button"
        >
          <Icon
            iconName="times"
          />
        </button>
      </span>
    </span>
    <span
      className="label label-dismissible label-secondary"
      key="4"
      tabIndex="-1"
    >
      <span
        className="label-item label-item-expand"
      >
        test
      </span>
      <span
        className="label-item label-item-after"
      >
        <button
          aria-label="Close"
          className="close"
          data-tag-index="4"
          onClick={[Function]}
          type="button"
        >
          <Icon
            iconName="times"
          />
        </button>
      </span>
    </span>
    <input
      className="form-control-inset"
      onKeyUp={[Function]}
      type="text"
    />
  </div>
  <div
    className="col-1 mt-1"
  >
    <Icon
      iconName="caret-double"
    />
  </div>
</div>
`;

exports[`Should hide the dropdown list 1`] = `
<div
  className=" align-items-start form-control form-control-tag-group multi-select-wrapper"
  onDoubleClick={[Function]}
  onFocus={[Function]}
>
  <div
    className="col-11 p-0"
  >
    <span
      className="label label-dismissible label-secondary"
      key="0"
      tabIndex="-1"
    >
      <span
        className="label-item label-item-expand"
      >
        ops
      </span>
      <span
        className="label-item label-item-after"
      >
        <button
          aria-label="Close"
          className="close"
          data-tag-index="0"
          onClick={[Function]}
          type="button"
        >
          <Icon
            iconName="times"
          />
        </button>
      </span>
    </span>
    <span
      className="label label-dismissible label-secondary"
      key="1"
      tabIndex="-1"
    >
      <span
        className="label-item label-item-expand"
      >
        message
      </span>
      <span
        className="label-item label-item-after"
      >
        <button
          aria-label="Close"
          className="close"
          data-tag-index="1"
          onClick={[Function]}
          type="button"
        >
          <Icon
            iconName="times"
          />
        </button>
      </span>
    </span>
    <span
      className="label label-dismissible label-secondary"
      key="2"
      tabIndex="-1"
    >
      <span
        className="label-item label-item-expand"
      >
        advocate
      </span>
      <span
        className="label-item label-item-after"
      >
        <button
          aria-label="Close"
          className="close"
          data-tag-index="2"
          onClick={[Function]}
          type="button"
        >
          <Icon
            iconName="times"
          />
        </button>
      </span>
    </span>
    <span
      className="label label-dismissible label-secondary"
      key="3"
      tabIndex="-1"
    >
      <span
        className="label-item label-item-expand"
      >
        Maybe
      </span>
      <span
        className="label-item label-item-after"
      >
        <button
          aria-label="Close"
          className="close"
          data-tag-index="3"
          onClick={[Function]}
          type="button"
        >
          <Icon
            iconName="times"
          />
        </button>
      </span>
    </span>
    <input
      className="form-control-inset"
      onKeyUp={[Function]}
      type="text"
    />
  </div>
  <div
    className="col-1 mt-1"
  >
    <Icon
      iconName="caret-double"
    />
  </div>
</div>
`;

exports[`Should remove a tag 1`] = `
<div
  className=" align-items-start form-control form-control-tag-group multi-select-wrapper"
  onDoubleClick={[Function]}
  onFocus={[Function]}
>
  <div
    className="col-11 p-0"
  >
    <span
      className="label label-dismissible label-secondary"
      key="0"
      tabIndex="-1"
    >
      <span
        className="label-item label-item-expand"
      >
        message
      </span>
      <span
        className="label-item label-item-after"
      >
        <button
          aria-label="Close"
          className="close"
          data-tag-index="0"
          onClick={[Function]}
          type="button"
        >
          <Icon
            iconName="times"
          />
        </button>
      </span>
    </span>
    <span
      className="label label-dismissible label-secondary"
      key="1"
      tabIndex="-1"
    >
      <span
        className="label-item label-item-expand"
      >
        advocate
      </span>
      <span
        className="label-item label-item-after"
      >
        <button
          aria-label="Close"
          className="close"
          data-tag-index="1"
          onClick={[Function]}
          type="button"
        >
          <Icon
            iconName="times"
          />
        </button>
      </span>
    </span>
    <span
      className="label label-dismissible label-secondary"
      key="2"
      tabIndex="-1"
    >
      <span
        className="label-item label-item-expand"
      >
        Maybe
      </span>
      <span
        className="label-item label-item-after"
      >
        <button
          aria-label="Close"
          className="close"
          data-tag-index="2"
          onClick={[Function]}
          type="button"
        >
          <Icon
            iconName="times"
          />
        </button>
      </span>
    </span>
    <input
      className="form-control-inset"
      onKeyUp={[Function]}
      type="text"
    />
  </div>
  <div
    className="col-1 mt-1"
  >
    <Icon
      iconName="caret-double"
    />
  </div>
</div>
`;

exports[`Should search 1`] = `
<div
  className=" align-items-start form-control form-control-tag-group multi-select-wrapper"
  onDoubleClick={[Function]}
  onFocus={[Function]}
>
  <div
    className="col-11 p-0"
  >
    <span
      className="label label-dismissible label-secondary"
      key="0"
      tabIndex="-1"
    >
      <span
        className="label-item label-item-expand"
      >
        message
      </span>
      <span
        className="label-item label-item-after"
      >
        <button
          aria-label="Close"
          className="close"
          data-tag-index="0"
          onClick={[Function]}
          type="button"
        >
          <Icon
            iconName="times"
          />
        </button>
      </span>
    </span>
    <span
      className="label label-dismissible label-secondary"
      key="1"
      tabIndex="-1"
    >
      <span
        className="label-item label-item-expand"
      >
        advocate
      </span>
      <span
        className="label-item label-item-after"
      >
        <button
          aria-label="Close"
          className="close"
          data-tag-index="1"
          onClick={[Function]}
          type="button"
        >
          <Icon
            iconName="times"
          />
        </button>
      </span>
    </span>
    <span
      className="label label-dismissible label-secondary"
      key="2"
      tabIndex="-1"
    >
      <span
        className="label-item label-item-expand"
      >
        Maybe
      </span>
      <span
        className="label-item label-item-after"
      >
        <button
          aria-label="Close"
          className="close"
          data-tag-index="2"
          onClick={[Function]}
          type="button"
        >
          <Icon
            iconName="times"
          />
        </button>
      </span>
    </span>
    <input
      className="form-control-inset"
      onKeyUp={[Function]}
      type="text"
    />
  </div>
  <div
    className="col-1 mt-1"
  >
    <Icon
      iconName="caret-double"
    />
  </div>
</div>
`;

exports[`Should show the dropdown list 1`] = `
<div
  className="is-active align-items-start form-control form-control-tag-group multi-select-wrapper"
  onDoubleClick={[Function]}
  onFocus={[Function]}
>
  <div
    className="col-11 p-0"
  >
    <span
      className="label label-dismissible label-secondary"
      key="0"
      tabIndex="-1"
    >
      <span
        className="label-item label-item-expand"
      >
        ops
      </span>
      <span
        className="label-item label-item-after"
      >
        <button
          aria-label="Close"
          className="close"
          data-tag-index="0"
          onClick={[Function]}
          type="button"
        >
          <Icon
            iconName="times"
          />
        </button>
      </span>
    </span>
    <span
      className="label label-dismissible label-secondary"
      key="1"
      tabIndex="-1"
    >
      <span
        className="label-item label-item-expand"
      >
        message
      </span>
      <span
        className="label-item label-item-after"
      >
        <button
          aria-label="Close"
          className="close"
          data-tag-index="1"
          onClick={[Function]}
          type="button"
        >
          <Icon
            iconName="times"
          />
        </button>
      </span>
    </span>
    <span
      className="label label-dismissible label-secondary"
      key="2"
      tabIndex="-1"
    >
      <span
        className="label-item label-item-expand"
      >
        advocate
      </span>
      <span
        className="label-item label-item-after"
      >
        <button
          aria-label="Close"
          className="close"
          data-tag-index="2"
          onClick={[Function]}
          type="button"
        >
          <Icon
            iconName="times"
          />
        </button>
      </span>
    </span>
    <span
      className="label label-dismissible label-secondary"
      key="3"
      tabIndex="-1"
    >
      <span
        className="label-item label-item-expand"
      >
        Maybe
      </span>
      <span
        className="label-item label-item-after"
      >
        <button
          aria-label="Close"
          className="close"
          data-tag-index="3"
          onClick={[Function]}
          type="button"
        >
          <Icon
            iconName="times"
          />
        </button>
      </span>
    </span>
    <input
      className="form-control-inset"
      onKeyUp={[Function]}
      type="text"
    />
  </div>
  <div
    className="col-1 mt-1"
  >
    <Icon
      iconName="caret-double"
    />
  </div>
  <div
    className="drop-suggestion dropdown-menu"
    tabIndex="-1"
  >
    <ul
      className="list-unstyled"
    >
      <li
        data-tag="meeting"
        key="0"
        onClick={[Function]}
      >
        <a
          className="dropdown-item "
          href="#0"
        >
          meeting
        </a>
      </li>
      <li
        data-tag="assist"
        key="1"
        onClick={[Function]}
      >
        <a
          className="dropdown-item "
          href="#1"
        >
          assist
        </a>
      </li>
      <li
        data-tag="not real"
        key="2"
        onClick={[Function]}
      >
        <a
          className="dropdown-item "
          href="#2"
        >
          not real
        </a>
      </li>
      <li
        data-tag="much"
        key="3"
        onClick={[Function]}
      >
        <a
          className="dropdown-item "
          href="#3"
        >
          much
        </a>
      </li>
      <li
        data-tag="so much"
        key="4"
        onClick={[Function]}
      >
        <a
          className="dropdown-item "
          href="#4"
        >
          so much
        </a>
      </li>
      <li
        data-tag="to do any"
        key="5"
        onClick={[Function]}
      >
        <a
          className="dropdown-item "
          href="#5"
        >
          to do any
        </a>
      </li>
      <li
        data-tag="a long component text example to test its overflow behavior."
        key="6"
        onClick={[Function]}
      >
        <a
          className="dropdown-item "
          href="#6"
        >
          a long component text example to test its overflow behavior.
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Should test component render 1`] = `
<div
  className=" align-items-start form-control form-control-tag-group multi-select-wrapper"
  onDoubleClick={[Function]}
  onFocus={[Function]}
>
  <div
    className="col-11 p-0"
  >
    <span
      className="label label-dismissible label-secondary"
      key="0"
      tabIndex="-1"
    >
      <span
        className="label-item label-item-expand"
      >
        ops
      </span>
      <span
        className="label-item label-item-after"
      >
        <button
          aria-label="Close"
          className="close"
          data-tag-index="0"
          onClick={[Function]}
          type="button"
        >
          <Icon
            iconName="times"
          />
        </button>
      </span>
    </span>
    <span
      className="label label-dismissible label-secondary"
      key="1"
      tabIndex="-1"
    >
      <span
        className="label-item label-item-expand"
      >
        message
      </span>
      <span
        className="label-item label-item-after"
      >
        <button
          aria-label="Close"
          className="close"
          data-tag-index="1"
          onClick={[Function]}
          type="button"
        >
          <Icon
            iconName="times"
          />
        </button>
      </span>
    </span>
    <span
      className="label label-dismissible label-secondary"
      key="2"
      tabIndex="-1"
    >
      <span
        className="label-item label-item-expand"
      >
        advocate
      </span>
      <span
        className="label-item label-item-after"
      >
        <button
          aria-label="Close"
          className="close"
          data-tag-index="2"
          onClick={[Function]}
          type="button"
        >
          <Icon
            iconName="times"
          />
        </button>
      </span>
    </span>
    <span
      className="label label-dismissible label-secondary"
      key="3"
      tabIndex="-1"
    >
      <span
        className="label-item label-item-expand"
      >
        Maybe
      </span>
      <span
        className="label-item label-item-after"
      >
        <button
          aria-label="Close"
          className="close"
          data-tag-index="3"
          onClick={[Function]}
          type="button"
        >
          <Icon
            iconName="times"
          />
        </button>
      </span>
    </span>
    <input
      className="form-control-inset"
      onKeyUp={[Function]}
      type="text"
    />
  </div>
  <div
    className="col-1 mt-1"
  >
    <Icon
      iconName="caret-double"
    />
  </div>
</div>
`;

exports[`Should test keydown press 1`] = `
<div
  className=" align-items-start form-control form-control-tag-group multi-select-wrapper"
  onDoubleClick={[Function]}
  onFocus={[Function]}
>
  <div
    className="col-11 p-0"
  >
    <span
      className="label label-dismissible label-secondary"
      key="0"
      tabIndex="-1"
    >
      <span
        className="label-item label-item-expand"
      >
        message
      </span>
      <span
        className="label-item label-item-after"
      >
        <button
          aria-label="Close"
          className="close"
          data-tag-index="0"
          onClick={[Function]}
          type="button"
        >
          <Icon
            iconName="times"
          />
        </button>
      </span>
    </span>
    <span
      className="label label-dismissible label-secondary"
      key="1"
      tabIndex="-1"
    >
      <span
        className="label-item label-item-expand"
      >
        advocate
      </span>
      <span
        className="label-item label-item-after"
      >
        <button
          aria-label="Close"
          className="close"
          data-tag-index="1"
          onClick={[Function]}
          type="button"
        >
          <Icon
            iconName="times"
          />
        </button>
      </span>
    </span>
    <span
      className="label label-dismissible label-secondary"
      key="2"
      tabIndex="-1"
    >
      <span
        className="label-item label-item-expand"
      >
        Maybe
      </span>
      <span
        className="label-item label-item-after"
      >
        <button
          aria-label="Close"
          className="close"
          data-tag-index="2"
          onClick={[Function]}
          type="button"
        >
          <Icon
            iconName="times"
          />
        </button>
      </span>
    </span>
    <input
      className="form-control-inset"
      onKeyUp={[Function]}
      type="text"
    />
  </div>
  <div
    className="col-1 mt-1"
  >
    <Icon
      iconName="caret-double"
    />
  </div>
</div>
`;

exports[`Should test keyenter press 1`] = `
<div
  className=" align-items-start form-control form-control-tag-group multi-select-wrapper"
  onDoubleClick={[Function]}
  onFocus={[Function]}
>
  <div
    className="col-11 p-0"
  >
    <span
      className="label label-dismissible label-secondary"
      key="0"
      tabIndex="-1"
    >
      <span
        className="label-item label-item-expand"
      >
        message
      </span>
      <span
        className="label-item label-item-after"
      >
        <button
          aria-label="Close"
          className="close"
          data-tag-index="0"
          onClick={[Function]}
          type="button"
        >
          <Icon
            iconName="times"
          />
        </button>
      </span>
    </span>
    <span
      className="label label-dismissible label-secondary"
      key="1"
      tabIndex="-1"
    >
      <span
        className="label-item label-item-expand"
      >
        advocate
      </span>
      <span
        className="label-item label-item-after"
      >
        <button
          aria-label="Close"
          className="close"
          data-tag-index="1"
          onClick={[Function]}
          type="button"
        >
          <Icon
            iconName="times"
          />
        </button>
      </span>
    </span>
    <span
      className="label label-dismissible label-secondary"
      key="2"
      tabIndex="-1"
    >
      <span
        className="label-item label-item-expand"
      >
        Maybe
      </span>
      <span
        className="label-item label-item-after"
      >
        <button
          aria-label="Close"
          className="close"
          data-tag-index="2"
          onClick={[Function]}
          type="button"
        >
          <Icon
            iconName="times"
          />
        </button>
      </span>
    </span>
    <span
      className="label label-dismissible label-secondary"
      key="3"
      tabIndex="-1"
    >
      <span
        className="label-item label-item-expand"
      >
        ops
      </span>
      <span
        className="label-item label-item-after"
      >
        <button
          aria-label="Close"
          className="close"
          data-tag-index="3"
          onClick={[Function]}
          type="button"
        >
          <Icon
            iconName="times"
          />
        </button>
      </span>
    </span>
    <input
      className="form-control-inset"
      onKeyUp={[Function]}
      type="text"
    />
  </div>
  <div
    className="col-1 mt-1"
  >
    <Icon
      iconName="caret-double"
    />
  </div>
</div>
`;

exports[`Should test keyup press 1`] = `
<div
  className=" align-items-start form-control form-control-tag-group multi-select-wrapper"
  onDoubleClick={[Function]}
  onFocus={[Function]}
>
  <div
    className="col-11 p-0"
  >
    <span
      className="label label-dismissible label-secondary"
      key="0"
      tabIndex="-1"
    >
      <span
        className="label-item label-item-expand"
      >
        message
      </span>
      <span
        className="label-item label-item-after"
      >
        <button
          aria-label="Close"
          className="close"
          data-tag-index="0"
          onClick={[Function]}
          type="button"
        >
          <Icon
            iconName="times"
          />
        </button>
      </span>
    </span>
    <span
      className="label label-dismissible label-secondary"
      key="1"
      tabIndex="-1"
    >
      <span
        className="label-item label-item-expand"
      >
        advocate
      </span>
      <span
        className="label-item label-item-after"
      >
        <button
          aria-label="Close"
          className="close"
          data-tag-index="1"
          onClick={[Function]}
          type="button"
        >
          <Icon
            iconName="times"
          />
        </button>
      </span>
    </span>
    <span
      className="label label-dismissible label-secondary"
      key="2"
      tabIndex="-1"
    >
      <span
        className="label-item label-item-expand"
      >
        Maybe
      </span>
      <span
        className="label-item label-item-after"
      >
        <button
          aria-label="Close"
          className="close"
          data-tag-index="2"
          onClick={[Function]}
          type="button"
        >
          <Icon
            iconName="times"
          />
        </button>
      </span>
    </span>
    <input
      className="form-control-inset"
      onKeyUp={[Function]}
      type="text"
    />
  </div>
  <div
    className="col-1 mt-1"
  >
    <Icon
      iconName="caret-double"
    />
  </div>
</div>
`;
